skstore/ts/src/skstore_api.ts

```diff
@@ -0,0 +1,588 @@
+   * @throws {Error} when an index contraints is broken
+   */
+  update(row: R, updates: JSONObject): void;
+  /**
+   * Update an entry in the table
+   * @param where - the column values to filter entries
+   * @param updates - the column values updates
+   * @throws {Error} when an index contraints is broken
+   */
+  updateWhere(where: JSONObject, updates: JSONObject): void;
+  /**
+   * Select entries in the table
+   * @param where - the column values to filter entries
+   * @throws {Error} when an index contraints is broken
+   */
+  select(where: JSONObject, columns?: string[]): JSONObject[];
+  /**
+   * Delete an entry in the table
+   * @param entry - the entry to delete from the table
+   */
+  delete(entry: R): void;
+  /**
+   * Delete entries in the table
+   * @param where - the column values to filter entries
+   */
+  deleteWhere(where: JSONObject): void;
+  /**
+   * Add a watch to the table
+   * @param update - the function to call on table update
+   */
+  watch: (update: (rows: JSONObject[]) => void) => { close: () => void };
+  /**
+   * Add a watch changes to the table
+   * @param init - the function to call on table init/reset
+   * @param update - the function to call on table update
+   */
+  watchChanges: (
+    init: (rows: JSONObject[]) => void,
+    update: (added: JSONObject[], removed: JSONObject[]) => void,
+  ) => { close: () => void };
+}
+
+export type Mapping<
+  K1 extends TJSON,
+  V1 extends TJSON,
+  K2 extends TJSON,
+  V2 extends TJSON,
+> = [EHandle<K1, V1>, Mapper<K1, V1, K2, V2>];
+
+export type Lazy<K extends TJSON, V extends TJSON> = (
+  selfHdl: LHandle<K, V>,
+  key: K,
+) => Opt<V>;
+
+export interface SKStoreFactory extends Shared {
+  runSKStore(
+    init: (skstore: SKStore, ...tables: TableHandle<TJSON[]>[]) => void,
+    app: App,
+    tables: MirrorSchema[],
+    connect?: boolean,
+  ): Promise<Table<TJSON[]>[]>;
+}
+
+export interface SKStore {
+  /**
+   * Creates a lazy reactive map
+   * @param {Lazy} compute - the lazy function to compute lazy map entry
+   * @returns {LHandle} The the resulting lazy reactive map handle
+   */
+  lazy<K extends TJSON, V extends TJSON>(compute: Lazy<K, V>): LHandle<K, V>;
+  /**
+   * Map over each entry of each eager reative map and apply the corresponding mapper function
+   * @param {Mapping} mappings - the handles to combine
+   * @returns {EHandle} The the resulting eager reactive map handle
+   */
+  multimap<
+    K1 extends TJSON,
+    V1 extends TJSON,
+    K2 extends TJSON,
+    V2 extends TJSON,
+  >(
+    mappings: Mapping<K1, V1, K2, V2>[],
+  ): EHandle<K2, V2>;
+  /**
+   * Map over each entry of each eager reative map and apply the corresponding mapper function
+   *  then reduce the when the given accumulator
+   * @param {Mapping} mappings - the handles to combine
+   * @param {Accumulator} accumulator - reduction manager
+   * @returns {EHandle} The the resulting eager reactive map handle
+   */
+  multimapReduce<
```

Comment:
Same comment as for `mapReduce` above: can we just define `reduce` directly and let users compose with multimap if/when they need?